src/Functions/toDayOfWeek.cpp

```diff
@@ -7,14 +7,38 @@
 namespace DB
 {
 
+namespace
```

Comment:
With my proposal (see my other comment), we should rather pass the mode argument as parameter to `ToDayOfWeekImpl` so that the "transformation" can be done on-the-fly.